packages/workspace/src/tasks-runner/task-orderer.ts

```diff
@@ -54,11 +118,18 @@ export class TaskOrderer {
     };
     tasks.forEach((t) => visitNode(t.target.project));
     const sortedTasks = [...tasks];
-    sortedTasks.sort((a, b) =>
-      sorted.indexOf(a.target.project) > sorted.indexOf(b.target.project)
-        ? 1
-        : -1
-    );
+    sortedTasks.sort((a, b) => {
```

Comment:
I reworked the topological sorting in master to be more straighforward. You should change your version to match that

Reply:
I do the following there:

```
    tasks.forEach((t) => visitNode(t.target.project));
    return sorted
      .map((projectName) => tasks.find((t) => t.target.project === projectName))
      .filter((t) => !!t);
 ```